Assets/Editor/GameAction.cs

```diff
@@ -1,41 +1,103 @@
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
 using UnityEditor;
 using UnityEngine;
 
-[CustomEditor(typeof(GameAction)), CanEditMultipleObjects]
-public class GameActionEditor : Editor {
+[CustomPropertyDrawer(typeof(GameAction))]
+public class GameActionEditor : PropertyDrawer {
 
-	public SerializedProperty
-		actionType,
-		damageValue,
-		healingValue;
+    public override void OnGUI(Rect position, SerializedProperty property, GUIContent label) {
+        EditorGUI.BeginProperty(position, label, property);
 
-	void OnEnable(){
-		// Set up all Serialized Properties here
-		actionType   = serializedObject.FindProperty("type");
-		damageValue  = serializedObject.FindProperty("damageValue");
-		healingValue = serializedObject.FindProperty("healingValue");
-	}
+        GameAction action = GetActionInstanceFromProperty(property);
+        if (action == null) {
+            return;
+        }
 
-	public override void OnInspectorGUI () {
-		serializedObject.Update();
+        position = EditorGUI.PrefixLabel(position, GUIUtility.GetControlID(FocusType.Passive), label);
 
-		// Always display the Action Type dropdown
-		EditorGUILayout.PropertyField( actionType );
+        var typeRect = new Rect(position.x, position.y, position.width, EditorGUIUtility.singleLineHeight);
+        var valueRect = new Rect(position.x, position.y + 20, position.width, EditorGUIUtility.singleLineHeight);
 
-		// Get the index of the selected Action Type
-		GameAction.ActionType type = (GameAction.ActionType)actionType.enumValueIndex;
+        EditorGUI.PropertyField(typeRect, property.FindPropertyRelative("type"));
+        switch (action.type) {
```

Comment:
There's a way to get the value of an enum through a serialized property: typeProp.enumValueIndex. And define the enums so casting will work without any problems:
``public enum ActionType {
  Invalid = 0,
  DamgeTarget = 1,
  HealTarget = 2
};``

This way could be more efficient than using Reflection and such.